module-services/service-appmgr/model/ApplicationManagerCommon.cpp

```diff
@@ -713,6 +717,7 @@ namespace app::manager
             setState(State::Running);
             EventManagerCommon::messageSetApplication(this, app.name());
             onLaunchFinished(app);
+            actionsRegistry.process();
```

Comment:
alright, so lets dive deeper.
It might be confusing, but `hasPendingAction` does not mean "is there an action in the queue we need to process" but rather "is there an action going on at the moment, which we are waiting to finish". And guess what, there isnt in this case.
That mechanism of launching an app where there is an action incoming **does not use Launch action**. During the application launch caused by incoming custom action **there is no "pending action"**. One might ask "shouldnt there be a pending action then?" but I insist on not reworking the whole action subsystem during emergency pre-release bugfix :)

So basically I agree that we can move the `actionRegistry.process()` deeper, ie to `onLaunchFinished`, but it still would have to be called before sanity `actionsRegistry.hasPendingAction()` and I see a little difference in bounding this call to either `onLaunchFinished` or `onSwitchConfirmed` - I can move it if you insist, let me know.

Reply:
I would at least insist to keep the mess in one place - therefore attach it to the function mentioned :)